### PR TITLE
level 90 MNK rotation

### PR DESCRIPTION
**Pull request by xanunderscore**
tested on normal and savage athena. also added positionals prediction, auto true north, and planner tracks.

this performs the TheBalance "braindead" rotation using the standard Lunar Solar opener and PR on odd minutes during Riddle of Fire. Optimal Drift is superior, especially for fights with downtime, but obviously a lot more complex to implement and will require testing and a followup PR when I get it working.

### CODE REVIEW

**Comment by Ants1988** [comment]
Any chance you could try dancer?

It's been requested in the past and would be great for us DNC players.

**Comment by xanunderscore** [comment]
I'm working on dancer atm. monk was easy to finish because most of the work is already done since there was a level 60 rotation implemented.

**Comment by Ants1988** [comment]
> I'm working on dancer atm. monk was easy to finish because most of the work is already done since there was a level 60 rotation implemented.

fantastic, good work so far!

File: BossMod/Autorotation/MNK/MNKActions.cs [hunk 1]
```diff
@@ -21,11 +20,21 @@ public Actions(Autorotation autorot, Actor player)
             _strategy = new();
 
             // upgrades
-            SupportedSpell(AID.SteelPeak).TransformAction = SupportedSpell(AID.ForbiddenChakra).TransformAction = () => ActionID.MakeSpell(_state.BestForbiddenChakra);
-            SupportedSpell(AID.HowlingFist).TransformAction = SupportedSpell(AID.Enlightenment).TransformAction = () => ActionID.MakeSpell(_state.BestEnlightenment);
-            SupportedSpell(AID.ArmOfTheDestroyer).TransformAction = SupportedSpell(AID.ShadowOfTheDestroyer).TransformAction = () => ActionID.MakeSpell(_state.BestShadowOfTheDestroyer);
-            SupportedSpell(AID.FlintStrike).TransformAction = SupportedSpell(AID.RisingPhoenix).TransformAction = () => ActionID.MakeSpell(_state.BestRisingPhoenix);
-            SupportedSpell(AID.TornadoKick).TransformAction = SupportedSpell(AID.PhantomRush).TransformAction = () => ActionID.MakeSpell(_state.BestPhantomRush);
+            SupportedSpell(AID.SteelPeak).TransformAction = SupportedSpell(
+                AID.ForbiddenChakra
```
Comment (awgil) [inline BossMod/Autorotation/MNK/MNKActions.cs:24]: Any reason for such formatting?

File: BossMod/Autorotation/MNK/MNKActions.cs [hunk 2]
```diff
@@ -94,13 +141,32 @@ private void UpdatePlayerState()
...
+            _state.BeastChakra = gauge.BeastChakra;
+            _state.Nadi = gauge.Nadi;
 
             (_state.Form, _state.FormLeft) = DetermineForm();
-            _state.DisciplinedFistLeft = StatusDetails(Player, SID.DisciplinedFist, Player.InstanceID).Left;
+            _state.DisciplinedFistLeft = StatusDetails(
+                Player,
+                SID.DisciplinedFist,
+                Player.InstanceID
+            ).Left;
```
[the remark this answers is not in the document]
Comment (xanunderscore) [reply BossMod/Autorotation/MNK/MNKActions.cs:154]: it's from CSharpier. I can't function without auto formatting, this is just how I am 😔 do you want me to revert the format changes that aren't on new code?

**Pull request merged**